src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -535,15 +557,99 @@ def _visit_neighbors(self, node: gtir.FunCall) -> DataExpr:
             memlet=dace.Memlet(data=neighbors_temp, subset=neighbor_idx),
         )
 
-        assert isinstance(node.type, itir_ts.ListType)
         return DataExpr(neighbors_node, node.type)
 
+    def _visit_map(self, node: gtir.FunCall) -> DataExpr:
+        assert isinstance(node.type, itir_ts.ListType)
+        assert isinstance(node.type.element_type, ts.ScalarType)
+        dtype = dace_utils.as_dace_type(node.type.element_type)
+
+        input_args = [self.visit(arg) for arg in node.args]
+
+        # TODO(edopao): extract offset_dim from the input arguments
+        offset_dim = gtx_common.Dimension("", gtx_common.DimensionKind.LOCAL)
+        map_index = dace_gtir_utils.get_map_variable(offset_dim)
+        connectors = [f"__arg{i}" for i in range(len(input_args))]
+
+        # The dataflow we build in this class has some loose connections on input edges.
+        # These edges are described as set of nodes, that will have to be connected to
+        # external data source nodes passing through the map entry node of the field map.
+        # For `map_` and `neighbors` expressions, these edges will terminate on the view
+        # nodes (see the for-loop below), because it is simpler than representing
+        # map-to-map edges (which require memlets with 2 pass-nodes).
+        input_memlets = {}
+        input_nodes = {}
+        local_size: Optional[int] = None
+        for conn, input_expr in zip(connectors, input_args, strict=True):
+            if isinstance(input_expr, MemletExpr):
+                if set(input_expr.subset.size()[:-1]) != {1}:
+                    raise ValueError(f"Invalid node {node}")
+                rstart, rstop, rstep = input_expr.subset[-1]
+                assert rstart == 0 and rstep == 1
+
+                desc = self.sdfg.arrays[input_expr.node.data]
+                view, _ = self.sdfg.add_view(
+                    f"{input_expr.node.data}_view",
+                    (rstop + 1,),
```

Comment:
why this shape and not `rstop - rstart`?

Reply:
I would add a note about that, because it is very confusing actually.

Reply:
I have changed the implementation, I am not using anymore `rstart` and `rstop`. Now I am using:
`view_shape = tuple(desc.shape[i] for i in local_dim_indices)`